deployment/administration/Delete_Unassigned_Users.ps1

```diff
@@ -0,0 +1,28 @@
+# Extract list of users
+$userOuPath = (Get-ADObject -Filter * | Where-Object { $_.Name -eq "Safe Haven Research Users" }).DistinguishedName
+$users = Get-ADUser -Filter * -SearchBase "$userOuPath" -Properties *
+foreach ($user in $users) {
+    $groupName = ($user | Select-Object -ExpandProperty MemberOf | ForEach-Object { (($_ -Split ",")[0] -Split "=")[1] }) -join "|"
+    $user | Add-Member -NotePropertyName GroupName -NotePropertyValue $groupName -Force
+}
+
+# Delete users not found in any group
+foreach ($user in $users) {
+    if (!($user.GroupName)) {
+        $name = $user.SamAccountName
+        Remove-ADUser -Identity $name
+    }
+}
```

Comment:
I don't love this pattern of getting an array of users and then decorating them with group name before iterating through them again. Why not just do:

```
for user in users
  if user not in group
    delete user
```